File: lawnchair/src/app/lawnchair/root/RootHelperBackend.java
package app.lawnchair.root;

import android.content.Context;
import android.os.PowerManager;
import android.os.RemoteException;
import android.os.SystemClock;

import androidx.core.content.ContextCompat;

public class RootHelperBackend extends IRootHelper.Stub {

    private final Context mContext;

    public RootHelperBackend(Context context) {
        mContext = context;
    }

    private PowerManager getPowerManager() {
        return ContextCompat.getSystemService(mContext, PowerManager.class);
    }

    @Override
    public void goToSleep() {
        getPowerManager().goToSleep(SystemClock.uptimeMillis());
    }
}

File: lawnchair/src/app/lawnchair/root/RootHelperBackend.kt
package app.lawnchair.root

import android.content.Context
import android.os.PowerManager
import android.os.SystemClock
import app.lawnchair.util.requireSystemService

class RootHelperBackend(private val mContext: Context) : IRootHelper.Stub() {
    private val powerManager: PowerManager get() = mContext.requireSystemService()

    override fun goToSleep() {
        powerManager.goToSleep(SystemClock.uptimeMillis())
    }
}
